### PR TITLE
fix(ui) Fix navigation between customer-domain accounts

### PR DESCRIPTION
**Pull request by markstory**
Improve navigation and subdomain support for `yarn dev-ui`. With these changes the following works:

- https://acme.localhost:7999/issues/
- https://acme.dev.getsentry.net:7999/issues/
- https://acme.abc1243.sentry.dev/issues/

Switching between organizations will also work, as domains will be rewritten based on the initial page configuration.

Users of dev-ui will still need to clone their session cookies and set host file entries if they want to use `acme.localhost` domains.

Refs HC-429

### CODE REVIEW

File: static/app/utils/useResolveRoute.tsx [hunk 1]
```diff
@@ -7,16 +7,37 @@ import {OrganizationContext} from 'sentry/views/organizationContext';
...
+ * or acme.dev.getsentry.net so we need to hack up the server provided
+ * domain values.
+ */
+function localizeDomain(domain?: string) {
+  if (!window.__SENTRY_DEV_UI || !domain) {
+    return domain;
+  }
+  const host = window.location.host;
+  const validHostPattern =
+    /((?:localhost|[^.]+\.sentry\.dev|dev\.getsentry\.net)(?:\:\d*)?)/;
```
Comment (evanpurkhiser) [inline static/app/utils/useResolveRoute.tsx:22]: Should this be some kind of constant, we have a similar regex above

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/utils/useResolveRoute.tsx (hunk 1 above)
Comment (markstory) [reply static/app/utils/useResolveRoute.tsx:22]: The capturing groups are different between the patterns though. The one in `useResolveRoute()` needs to extract the slug, while this function needs to match the trailing domain + port.

I can try replacing the regex patterns with split on `.` operations.

**Review by priscilawebdev — APPROVED** [review]
code looks good to me. Thanks for fixing it @markstory 🥔 🙏

**Comment by priscilawebdev** [comment]
I can confirm that `https://acme.localhost:7999/issues/` works...but it's no longer possible to switch organizations anymore because of the domain change... is there anything we can do to make this possible again?

**Comment by markstory** [comment]
> is there anything we can do to make this possible again?

Yes when you clone your cookies into localhost make sure to set the domain attribute to `localhost` and not `acme.localhost`. That will allow the cookie to be shared across all of the subdomains.

**Pull request merged**